### PR TITLE
include keymanager runtime, use release artifacts for docker image build

### PR DESCRIPTION
**Pull request by ptrus**
Closes: https://github.com/oasislabs/oasis-runtime/issues/937

Done:
- [x] depend on oasis-core 20.5 release
- [x] add keymanager runtime crate (also see: https://github.com/oasislabs/oasis-core/pull/2837)
- [x] fix docker image to be built from artifacts
- [x] test CI works
- [x] test image build CI & bencharmks run with built image

### CODE REVIEW

File: .buildkite/deploy.pipeline.yml [hunk 1]
```diff
@@ -27,46 +27,14 @@ docker_plugin: &docker_plugin_configuration
...
   - label: Get docker tag and save it as metadata for use later
     branches: master
     command: .buildkite/scripts/set_docker_tag_meta_data.sh
 
   - label: Build testing artifacts
-    branches: master devnet2
-    command:
-      - .buildkite/scripts/build_deployment_context.sh platform context.tar.gz
+    branches: master
+    command: # TODO: extract version.
```
Comment (kostko) [inline .buildkite/deploy.pipeline.yml:36]: How will the version be extracted?

Comment (ptrus) [reply .buildkite/deploy.pipeline.yml:36]: once we can specify a vesrion we could parse cargo.toml? until then we can probably a file containing the version somewhere in the repo

Comment (ptrus) [reply .buildkite/deploy.pipeline.yml:36]: Something like a `.oasis_core_version` file in the repo root. Seems better than specifying the version in X places.

Comment (kostko) [reply .buildkite/deploy.pipeline.yml:36]: Yeah sounds reasonable.

File: gateway/Cargo.toml [hunk 2]
```diff
@@ -34,12 +34,18 @@ mem = { git = "https://github.com/oasislabs/oasis-parity", branch = "ekiden" }
 ## Common functionality between runtime and gateway
 oasis-runtime-common = { path = "../common" }
 
-## Oasis Core client dependencies
-clap = "2.29.1"
-oasis-core-client = { git = "https://github.com/oasislabs/oasis-core", branch = "master" }
-oasis-core-runtime = { git = "https://github.com/oasislabs/oasis-core", branch = "master" }
-oasis-core-keymanager-client = { git = "https://github.com/oasislabs/oasis-core", branch = "master" }
+## Oasis Core client dependencies.
+# The referenced commit has rust code compatible with artifacts in oasis-core
```
Comment (peterjgilbert) [inline gateway/Cargo.toml:38]: Can remove this comment now

File: genesis/Cargo.toml [hunk 3]
```diff
@@ -5,8 +5,15 @@ authors = ["Oasis Labs Inc. <info@oasislabs.com>"]
 
 [dependencies]
 oasis-runtime-common = { path = "../common" }
-oasis-core-client = { git = "https://github.com/oasislabs/oasis-core", branch = "master" }
-oasis-core-runtime = { git = "https://github.com/oasislabs/oasis-core", branch = "master" }
+
+# The referenced commit has rust code compatible with artifacts in oasis-core
```
Comment (peterjgilbert) [inline genesis/Cargo.toml:9]: Can remove this comment now

File: keymanager-runtime/Cargo.toml [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+authors = ["Oasis Labs Inc. <info@oasislabs.com>"]
+edition = "2018"
+
+[package.metadata.fortanix-sgx]
+heap-size = 134217728
+stack-size = 2097152
+threads = 2
+
+[dependencies]
+# The referenced commit has rust code compatible with artifacts in oasis-core
```
Comment (peterjgilbert) [inline keymanager-runtime/Cargo.toml:13]: Remove comment

Comment (ptrus) [reply keymanager-runtime/Cargo.toml]: done :+1: merging this since CI passed, and only changes after it were removing the comments

**Pull request merged**